src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java

```diff
@@ -0,0 +1,493 @@
+    /** A newline with 14 spaces of indentation. */
+    private static final String INDENT_LEVEL_14 = SiteUtil.getNewlineAndIndentSpaces(14);
+    /** A newline with 16 spaces of indentation. */
+    private static final String INDENT_LEVEL_16 = SiteUtil.getNewlineAndIndentSpaces(16);
+    /** A newline with 18 spaces of indentation. */
+    private static final String INDENT_LEVEL_18 = SiteUtil.getNewlineAndIndentSpaces(18);
+    /** A newline with 20 spaces of indentation. */
+    private static final String INDENT_LEVEL_20 = SiteUtil.getNewlineAndIndentSpaces(20);
+
+    /** Path to main source code folder. */
+    private static final String MAIN_FOLDER_PATH = Paths.get(
+            "src", "main", "java", "com", "puppycrawl", "tools", "checkstyle").toString();
+
+    /** List of files who are parent modules and contain certain properties that checks inherit. */
+    private static final List<File> PARENT_MODULES_WITH_PROPERTIES = List.of(
+        new File(Paths.get(MAIN_FOLDER_PATH,
+                CHECKS, NAMING, "AbstractAccessControlNameCheck.java").toString()),
+        new File(Paths.get(MAIN_FOLDER_PATH,
+                CHECKS, NAMING, "AbstractNameCheck.java").toString()),
+        new File(Paths.get(MAIN_FOLDER_PATH,
+                CHECKS, "javadoc", "AbstractJavadocCheck.java").toString()),
+        new File(Paths.get(MAIN_FOLDER_PATH,
+                "api", "AbstractFileSetCheck.java").toString()),
+        new File(Paths.get(MAIN_FOLDER_PATH,
+                CHECKS, "header", "AbstractHeaderCheck.java").toString())
+    );
+
+    /** Mao of all parent modules properties and their javadocs. */
+    private static final Map<String, DetailNode> PARENT_MODULES_PROPERTIES_JAVADOCS =
+            new HashMap<>();
+
+    /** The name of the current module being processed. */
+    private static String currentModuleName = "";
+
+    @Override
+    public void execute(Sink sink, MacroRequest request) throws MacroExecutionException {
+        // until https://github.com/checkstyle/checkstyle/issues/13426
+        if (!(sink instanceof XdocSink)) {
+            throw new MacroExecutionException("Expected Sink to be an XdocSink.");
+        }
+
+        // lazy initialization
+        if (PARENT_MODULES_PROPERTIES_JAVADOCS.isEmpty()) {
+            processParentModules();
+        }
+
+        final String modulePath = (String) request.getParameter("modulePath");
+        final File moduleFile = new File(modulePath);
+        currentModuleName = CommonUtil.getFileNameWithoutExtension(moduleFile.getName());
+
+        processModule(currentModuleName, moduleFile);
+        writePropertiesTable((XdocSink) sink);
+    }
+
+    /**
+     * Collect the properties setters javadocs of the parent modules.
+     *
+     * @throws MacroExecutionException if an error occurs during processing.
+     */
+    private static void processParentModules() throws MacroExecutionException {
```

Comment:
We need to revisit usage of `parent` and `parentModule` throughout this entire PR, this is very confusing. Take `RecordCompoenentNumberCheck` . It has a parent (`TreeWalker`) and a super class: `AbstractCheck`.

Reply:
Done.